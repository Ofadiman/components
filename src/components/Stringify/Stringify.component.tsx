import React from 'react'

import { StyledStringify } from './Stringify.styles'
import { StringifyProps } from './Stringify.types'

export const Stringify: FC<StringifyProps> = ({ className, value, indentBy = 2 }) => (
  <StyledStringify className={className}>{JSON.stringify(value, null, indentBy)}</StyledStringify>
)
