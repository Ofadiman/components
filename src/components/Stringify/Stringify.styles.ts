import styled, { css } from 'styled-components'

export const StyledStringify = styled.pre`
  ${({ theme }) => css`
    background-color: ${theme.palette.primary.main};
    border-radius: ${theme.bordering(4)};
    box-shadow: ${theme.shadowing(2)};
    color: ${theme.palette.primary.contrastText};
    font-size: 2rem;
    padding: ${theme.spacing(8)};
    width: fit-content;
  `}
`
