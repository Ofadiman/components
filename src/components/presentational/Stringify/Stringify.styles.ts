import { makeStyles } from '@material-ui/core/styles'

export const useStringifyStyles = makeStyles(
  () => ({
    root: {
      fontSize: '1.6rem'
    }
  }),
  {
    name: 'Stringify'
  }
)
