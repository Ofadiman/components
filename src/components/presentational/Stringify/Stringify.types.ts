type StringifyValue = string | number | object | null

type StringifyArray = (StringifyValue | StringifyArray)[]

export interface StringifyProps {
  className?: string
  indentBy?: number
  value: StringifyValue | StringifyArray
}
