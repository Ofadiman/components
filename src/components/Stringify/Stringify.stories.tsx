import { storiesOf } from '@storybook/react'
import React from 'react'

import { Stringify } from './Stringify.component'

storiesOf('Stringify', module).add('Default', () => (
  <Stringify value={[{ foo: { bar: 'buzz' } }, null, 1, 'string', ['nest', 'array']]} />
))
