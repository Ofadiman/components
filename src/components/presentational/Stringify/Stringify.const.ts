import { ComponentStoriesProps } from '../../../types/ComponentStoriesProps.type'
import { StringifyProps } from './Stringify.types'

export const stringifyStoryProps: ComponentStoriesProps<StringifyProps> = [
  {
    title: 'Default',
    value: [{ foo: { bar: 'buzz' } }, null, 1, 'string', ['nest', 'array']]
  },
  {
    indentBy: 8,
    title: 'Custom indentation',
    value: [{ foo: { bar: 'buzz' } }, null, 1, 'string', ['nest', 'array']]
  }
]
