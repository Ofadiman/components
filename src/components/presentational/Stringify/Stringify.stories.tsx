import { storiesOf } from '@storybook/react'
import React from 'react'

import { Stringify } from './Stringify'
import { stringifyStoryProps } from './Stringify.const'

const clientApi = storiesOf('Components/Presentational/Stringify', module)

stringifyStoryProps.forEach(({ title, ...props }) => {
  clientApi.add(title, () => <Stringify {...props} />)
})
