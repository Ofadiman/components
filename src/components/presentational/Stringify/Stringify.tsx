import React from 'react'

import { StringifyProps } from './Stringify.types'

export const Stringify: FC<StringifyProps> = ({ className, indentBy = 4, value }) => {
  return <pre className={className}>{JSON.stringify(value, null, indentBy)}</pre>
}
