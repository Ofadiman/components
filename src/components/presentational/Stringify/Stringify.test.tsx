import React from 'react'

import { renderComponent } from '../../../functions/renderComponent/renderComponent'
import { Stringify } from './Stringify'
import { stringifyStoryProps } from './Stringify.props'

describe('Stringify component', () => {
  stringifyStoryProps.forEach(({ title, ...props }) => {
    test(`match snapshot for ${title}`, () => {
      const { asFragment } = renderComponent(<Stringify {...props} />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
