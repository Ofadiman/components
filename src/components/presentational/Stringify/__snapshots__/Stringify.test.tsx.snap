// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stringify component match snapshot for Custom indentation 1`] = `
<DocumentFragment>
  <pre>
    [
        {
                "foo": {
                        "bar": "buzz"
                }
        },
        null,
        1,
        "string",
        [
                "nest",
                "array"
        ]
]
  </pre>
</DocumentFragment>
`;

exports[`Stringify component match snapshot for Default 1`] = `
<DocumentFragment>
  <pre>
    [
    {
        "foo": {
            "bar": "buzz"
        }
    },
    null,
    1,
    "string",
    [
        "nest",
        "array"
    ]
]
  </pre>
</DocumentFragment>
`;
